File: T3/proyectos/ExamenCorregido/app/src/main/java/com/borja/examencorregido/SecondActivity.java
package com.borja.examencorregido;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

public class SecondActivity extends AppCompatActivity {

    TextView tRecupera;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        instancias();
        recuperarDatos();
    }

    private void recuperarDatos() {
        if (getIntent().getExtras()!=null){

            String tipo = (String) getIntent().getExtras().get(MainActivity.TAG2);
            int resultado= (int) getIntent().getExtras().get(MainActivity.TAG1);

            tRecupera.setText(String.format("La operación de %s es %d",tipo,resultado));
        }
    }

    private void instancias() {
        tRecupera = findViewById(R.id.texto_recibido);
    }
}

File: Kotlin/T3/codigos_ces/Concesionario/app/src/main/java/com/example/concesionario/ui/SecondActivity.kt
package com.example.concesionario.ui

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.concesionario.R
import com.example.concesionario.adapter.AdapterModelos
import com.example.concesionario.databinding.ActivitySecondBinding
import com.example.concesionario.model.Marca
import com.example.concesionario.model.Vehiculo

class SecondActivity : AppCompatActivity() {

    lateinit var binding: ActivitySecondBinding
    lateinit var adapterModelos: AdapterModelos
    lateinit var listaModelos: ArrayList<Vehiculo>
    var vehiculo: String? = null
    var marca: Marca? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySecondBinding.inflate(layoutInflater)
        setContentView(binding.root)
        instancias()
        recuperarDatos()
        initGUI()
    }

    private fun instancias() {
        listaModelos = arrayListOf(
            Vehiculo(
                Marca("Fiat", R.drawable.fiat),
                "600",
                100,
                20000,
                "rojo",
                R.drawable.fiats
            ),
            Vehiculo(
                Marca("Ford", R.drawable.ford),
                "Focus",
                150,
                30000,
                "azul",
                R.drawable.ford
            )
        )
        adapterModelos = AdapterModelos(listaModelos,this)
    }

    private fun initGUI() {
        binding.textoVehiculo.setText(vehiculo)
        binding.textoMarca.setText(marca?.nombre)
        binding.spinnerModelos.adapter = adapterModelos
    }

    private fun recuperarDatos() {
        vehiculo = intent.getStringExtra("vehiculo")
        marca = intent.getSerializableExtra("marca") as Marca
    }
}